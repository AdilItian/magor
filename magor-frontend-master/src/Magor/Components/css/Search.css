.searchBar {
    height: 67px;
    padding: 10px 30px;
    font-size: 40px;
    border-radius: 8px;
    border: 1px solid #808080;
    width: 50vw;
    min-width: 300px;
}

.searchBar.small {
    height: 47px;
    padding: 20px;
    font-size: 28px;
    border-radius: 6px;
    border: 1px solid #808080;
    width: 50vw;
    min-width: 300px;
}

.searchButton {
    height: 67px;
    width: 67px;
    border: none;
    border-radius: 0 8px 8px 0;
    background-color: #1faeff;
    background-image: url(/search.svg);
    background-position: center;
    background-size: 50px 50px;
    background-repeat: no-repeat;
    color: white;
    right: 0;
    position: absolute;
    cursor: pointer;
}

.searchButton.small {
    height: 47px;
    width: 47px;
    background-size: 32px;
}

.searchBarContainer {
    position: relative;
}

.searchBarContainer ul {
    display: none;
    position: absolute;
    background: #fff;
    top: 100%;
    margin: 0;
    border: 3px solid #1faeff;
    border-top: none;
    border-radius: 0 0 10px 10px;
    list-style: none;
    padding: 0;
    max-height: 165px;
    overflow-y: scroll;
}

.searchBar:focus+ul,
.searchBarContainer ul:hover {
    display: block;
}

.searchBarContainer li {
    padding: 5px 10px;
    cursor: pointer;
}

.searchBarContainer li:hover {
    background: #1faeff;
    color: #fff;
    font-weight: bold;
    ;
}