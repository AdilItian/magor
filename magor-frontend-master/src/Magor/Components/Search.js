import React from 'react'
import { DropDown } from './Inputs'
import './css/Search.css'
import cx from '../Utils/cx'

export const SearchBar = (props) => {
    const handleKeyUp = (e) => {
        if (e.keyCode === 13) {
            props.performSearch()
        }
    }
    const matches = (props.matches || []).filter((m) => m !== props.value)
    if (typeof props.performSearch !== 'function') return null
    if (typeof props.handleChange !== 'function') return null
    return (
        <div className="searchBarContainer">
            <input
                type="text"
                name="search"
                placeholder={props.placeholder || 'Enter search query here'}
                onKeyUp={handleKeyUp}
                onChange={(e) => props.handleChange(e.target.value)}
                value={props.value || ''}
                className={`font searchBar ${props.small ? 'small' : ''}`}
                style={props.style}
            />
            {matches.length > 0 && (
                <ul>
                    {matches.map((li) => (
                        <li
                            key={li}
                            onClick={(_) => {
                                props.handleChange(li)
                                props.performSearch(li)
                            }}
                        >
                            {li}
                        </li>
                    ))}
                </ul>
            )}
        </div>
    )
}

export const SearchButton = (props) => {
    return typeof props.performSearch !== 'function' ? null : (
        <button
            onClick={props.performSearch}
            className={`searchButton ${props.small ? 'small' : ''}`}
            style={props.style}
        />
    )
}

export const NamedSearchBar = (props) => {
    return (
        <>
            <SearchBar
                value={props.value}
                performSearch={props.performSearch}
                handleChange={props.handleChange}
                small={props.small}
            />
            <div
                id="SearchName"
                className={cx({
                    hidden: props.value === '',
                    small: props.small,
                })}
            >
                <DropDown
                    small={props.small}
                    selected={props.name}
                    list={props.names}
                    setSelected={props.setName}
                />
            </div>
        </>
    )
}
