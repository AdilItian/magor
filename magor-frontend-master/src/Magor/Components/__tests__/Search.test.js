import React from 'react'
import { SearchBar, SearchButton, NamedSearchBar } from '../Search'
import renderer, { act } from 'react-test-renderer'

test('No undefined', () => {
    expect(SearchBar).toBeDefined()
    expect(SearchButton).toBeDefined()
    expect(NamedSearchBar).toBeDefined()
})

test('SearchButton basic tests', () => {
    const searchButton = renderer.create(<SearchButton />)
    expect(searchButton.toJSON()).toBeNull()

    const searchButtonWithEmptyFunc = renderer.create(
        <SearchButton performSearch={1} />
    )
    expect(searchButtonWithEmptyFunc.toJSON()).toBeNull()

    const searchButtonWithFunc = renderer.create(
        <SearchButton performSearch={() => {}} />
    )
    expect(searchButtonWithFunc.toJSON()).toMatchSnapshot()
    expect(searchButtonWithFunc.toJSON().props.className.trim()).toBe(
        'searchButton'
    )

    const searchButtonSmall = renderer.create(
        <SearchButton small performSearch={() => {}} />
    )
    expect(searchButtonSmall.toJSON()).toMatchSnapshot()
    expect(searchButtonSmall.toJSON().props.className.trim()).toBe(
        'searchButton small'
    )
})

test('SearchButton calls callback on click', () => {
    const performSearch = jest.fn(() => {})
    const searchButton = renderer.create(
        <SearchButton performSearch={performSearch} />
    )
    let tree = searchButton.toJSON()
    expect(tree).toMatchSnapshot()
    tree.props.onClick()
    expect(performSearch).toHaveBeenCalledTimes(1)
})

test('Basic SearchBar tests', () => {
    const searchBarWithoutCallbacks = renderer.create(<SearchBar />)
    expect(searchBarWithoutCallbacks.toJSON()).toBeNull()

    const searchBarWithoutHandleChange = renderer.create(
        <SearchBar performSearch={() => {}} />
    )
    expect(searchBarWithoutHandleChange.toJSON()).toBeNull()

    const searchBarWithoutPerformSearch = renderer.create(
        <SearchBar handleChange={() => {}} />
    )
    expect(searchBarWithoutPerformSearch.toJSON()).toBeNull()

    const validSearchBar = renderer.create(
        <SearchBar handleChange={() => {}} performSearch={() => {}} />
    )
    expect(validSearchBar.toJSON()).toMatchSnapshot()

    const searchBarSmall = renderer.create(
        <SearchBar small handleChange={() => {}} performSearch={() => {}} />
    )
    expect(searchBarSmall.toJSON()).toMatchSnapshot()
    expect(searchBarSmall.root.findByType('input').props.className.trim()).toBe(
        'font searchBar small'
    )

    const searchBarWithMatches = renderer.create(
        <SearchBar
            matches={['match1', 'match2']}
            handleChange={() => {}}
            performSearch={() => {}}
        />
    )
    expect(searchBarWithMatches.root.findAllByType('li')).toHaveLength(2)
    expect(
        searchBarWithMatches.root
            .findAllByType('li')
            .map((li) => li.children[0])
    ).toEqual(['match1', 'match2'])
    expect(searchBarWithMatches.toJSON()).toMatchSnapshot()
})

test('Searchbar functionality tests', () => {
    const handleChange = jest.fn(() => {})
    const performSearch = jest.fn(() => {})
    const searchbar = renderer.create(
        <SearchBar handleChange={handleChange} performSearch={performSearch} />
    )
    searchbar.root
        .findByType('input')
        .props.onChange({ target: { value: 'Entered text' } })
    expect(handleChange).toHaveBeenCalledTimes(1)
    expect(handleChange.mock.calls[0][0]).toBe('Entered text')
    searchbar.root.findByType('input').props.onKeyUp({ keyCode: 13 })
    expect(performSearch).toHaveBeenCalledTimes(1)
    searchbar.root
        .findByType('input')
        .props.onKeyUp({ keyCode: 'anything else' })
    expect(performSearch).toHaveBeenCalledTimes(1)
})

test('Matches onclick test', () => {
    const handleChange = jest.fn(() => {})
    const performSearch = jest.fn(() => {})
    const searchbar = renderer.create(
        <SearchBar
            handleChange={handleChange}
            performSearch={performSearch}
            matches={['match1', 'match2']}
        />
    )
    searchbar.root.findAllByType('li')[0].props.onClick()
    expect(handleChange).toHaveBeenCalledTimes(1)
    expect(performSearch).toHaveBeenCalledTimes(1)
    expect(handleChange.mock.calls[0][0]).toBe('match1')
    expect(performSearch.mock.calls[0][0]).toBe('match1')
})

test('Named search Bar', async () => {
    let searchbar
    const value = 'test'
    const emptyFunction = jest.fn(() => {})
    const name = 'a'
    const names = ['a', 'b', 'c']
    await act(async () => {
        searchbar = renderer.create(
            <NamedSearchBar
                value={value}
                name={name}
                names={names}
                setValue={emptyFunction}
                setName={emptyFunction}
                performSearch={emptyFunction}
                handleChange={emptyFunction}
            />
        )
    })
    expect(searchbar).toMatchSnapshot()
})
