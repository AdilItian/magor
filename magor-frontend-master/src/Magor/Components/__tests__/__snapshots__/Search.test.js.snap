// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic SearchBar tests 1`] = `
<div
  className="searchBarContainer"
>
  <input
    className="font searchBar "
    name="search"
    onChange={[Function]}
    onKeyUp={[Function]}
    placeholder="Enter search query here"
    type="text"
  />
</div>
`;

exports[`Basic SearchBar tests 2`] = `
<div
  className="searchBarContainer"
>
  <input
    className="font searchBar small"
    name="search"
    onChange={[Function]}
    onKeyUp={[Function]}
    placeholder="Enter search query here"
    type="text"
  />
</div>
`;

exports[`Basic SearchBar tests 3`] = `
<div
  className="searchBarContainer"
>
  <input
    className="font searchBar "
    name="search"
    onChange={[Function]}
    onKeyUp={[Function]}
    placeholder="Enter search query here"
    type="text"
  />
  <ul>
    <li
      onClick={[Function]}
    >
      match1
    </li>
    <li
      onClick={[Function]}
    >
      match2
    </li>
  </ul>
</div>
`;

exports[`Named search Bar 1`] = `
Array [
  <div
    className="searchBarContainer"
  >
    <input
      className="font searchBar "
      name="search"
      onChange={[Function]}
      onKeyUp={[Function]}
      placeholder="Enter search query here"
      type="text"
      value="test"
    />
  </div>,
  <div
    className=""
    id="SearchName"
  >
    <div
      className="dropDown"
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      style={
        Object {
          "width": 220,
        }
      }
    >
      <p>
        a
      </p>
      <ul>
        <li
          onClick={[Function]}
        >
          b
        </li>
        <li
          onClick={[Function]}
        >
          c
        </li>
      </ul>
    </div>
  </div>,
]
`;

exports[`SearchButton basic tests 1`] = `
<button
  className="searchButton "
  onClick={[Function]}
/>
`;

exports[`SearchButton basic tests 2`] = `
<button
  className="searchButton small"
  onClick={[Function]}
/>
`;

exports[`SearchButton calls callback on click 1`] = `
<button
  className="searchButton "
  onClick={[MockFunction]}
/>
`;
